ccmbioinfo/lfs
# CARDS-1867: As an admin, I can configure the patient identification process

[Jira link](https://phenotips.atlassian.net/browse/CARDS-1867)

In implementing CARDS-1867, this also covers [CARDS-1771](https://phenotips.atlassian.net/browse/CARDS-1771), [CARDS-1772](https://phenotips.atlassian.net/browse/CARDS-1772), and [CARDS-1845](https://phenotips.atlassian.net/browse/CARDS-1845).

This commit adds a new administration page, Patient Identification, which changes how the Proms homepage works.
![image](https://user-images.githubusercontent.com/4656440/185635819-e0bf89a0-1738-44e4-a658-df367d27225d.png)

1. `Patients can answer surveys without a personalized link` will change the default login page, preventing people from logging in as a patient when they enter it.

2. `Patients must confirm their identity by providing their date of birth and either MRN or HCN` will control the patient login page. This can only be turned off if users are required to use a personalized link (i.e. 1. is disabled), as without a token we otherwise have no means of determining which patient a user is.

3. `The authentication token is valid after the associated event for:` alters how long authentication tokens provided by the email service are valid for. Prior to this PR, they were valid until midnight the day of the event. They are now set to 0 days after the event (the night of) by default.

*To test*:
1. While logged in as admin, you can alter Patient Identification settings from Administration/Patient Identification on the sidebar.
2. You can confirm the behaviour of 1. by altering it and logging out. The patient login will no longer be available.
3. You can confirm the behaviour of 2. by generating a patient login token. To do this, create a Patient Information form and a Visit Information form (or import some from the Proms importer using the instructions at #979 ). Then, try logging in with the token by visiting localhost:8080/Proms/?auth_token=<token>.
4. You can confirm the behaviour of 3. by forcing an email to be sent (see [these instructions](https://github.com/data-team-uhn/cards/blob/dev/Utilities/Development/EmailServer/README.md)), and checking the expiry date of the token generated. Easiest way to look at all tokens generated is by visiting http://localhost:8080/jcr:system/cards:tokens.2.json .
